src/top_level_parser.rs

```diff
@@ -25,7 +31,49 @@ enum TopLevelStat {
     Memused(u64),
     MemusedMax(u64),
     LnetMemused(u64),
-    HealthCheck(String),
+    HealthCheck(HealthCheckStat),
+}
+
+fn target_health<I>() -> impl Parser<I, Output = Target>
+where
+    I: Stream<Token = char>,
+    I::Error: ParseError<I::Token, I::Range, I::Position>,
+{
+    (
+        string("device"),
+        space(),
+        target(),
+        space(),
+        string("reported"),
+        space(),
+        words(),
```

Comment:
You are treating "health" as `words`. I'd feel more comfortable if you matched on the string explicitly.

You also should be sure that health only has that exact string as a posibility.

Reply:
Make sense, looks like it's always `unhealthy`
https://github.com/whamcloud/lustre/blob/a336d7c7c1cd62a5a5213835aa85b8eaa87b076a/lustre/obdclass/obd_sysfs.c#L524-L526